03_building_and_packaging/cpack_exercise.md

```diff
@@ -1,118 +1,66 @@
+The same four steps as in the lecture:
 
-The `CMakeLists.txt` should look familiar. It has two targets
+### 1. Add Install Target to CMake Configuration
 
-1. `cpackexample`: This target creates an executable with same name.
-2. `cpackexamplelib`: This target creates a library (called `libcpackexamplelib.so` or `libcpackexamplelib.a`).
+- The executable `cpackexample` should be installed in a `<prefix>/bin/` directory, the library `libcpackexamplelib.a/so` should be installed in a `<prefix>/lib/` directory, and all header files (`fem.hpp`, `filesystem.hpp`, `flatset.hpp`, `yamlParser.hpp`) should be installed in a `<prefix>/include/cpackexamplelib` directory.
+- Test whether `make install` works as expected.
 
-The executable will take one command line argument to specify a `yaml` file to be parsed. This allows to parse the file `config.yml` inside the `yamlParser/` directory even if the executable is not located within the repository anymore. In last week's exercise the path to `config.yaml` was hard coded. In this exercise you can call the program as
+### 2. Add CPack Configuration
 
-```bash
-cpackexample <path_to_repo>/yamlParser/config.yml
-```
+- Write a seperate CMake module `cmake/CPackConfig.cmake` for the packaging process and include it in the `CMakeLists.txt` file. The created package should contain sufficient information about the package, at least: maintainer, contact, project description, vendor, and homepage (e.g. your fork on GitHub). Feel free to set more [additional options](https://cmake.org/cmake/help/latest/module/CPack.html).
+- `make package` should (only) create a `tar.gz` and a `deb` package.
+- Inspect that both packages contain the correct content:
+    - To inspect the contents of a `.tar.gz.` file, you can unpack it using the tool `tar`:
 
-If you start the program without the additional argument, the program will not try to parse anything.
+      ```bash
+      tar -xzf TARGZFILE
+      ```
 
-### Development/Packaging Environment
+    - To inspect the contents of a  `.deb` file, you can unpack it using the tool `dpkg-deb`:
 
-Please work inside a Docker container when packaging the code. The image should be based on *Ubuntu 20.04* such that you have a consistent work environment. Such a container was also used when the exercise was prepared.
+      ```bash
+      dpkg-deb -R DEBFILE DIRECTORY_FOR_UNPACKED_DEBFILE
+      ```
 
-The code has (almost) the same dependencies as last week, Therefore, you can reuse your container most parts from the previous week's exercise's container, but install `libyaml-cpp-dev` via `apt` instead of compiling `yaml-cpp` manually.
+- Optional: Do the dependencies look correct? Modify the Dockerfile such that `libyaml-cpp` also properly appears as dependency.
```

Comment:
Checking the dependencies only makes sense after step 3 is done, as at this point no dependency setting is done in CPack.